Prevent the resize handle from covering the center

matthewwithanm opened the pull request:
Currently, the absolutely positioned element is the only thing with a
size. That means that it'll cover the center. With this change, we give
a size to the (relatively-positioned) atom-dock element so that closed
docks take up canvas space instead.

I think @simurai mentioned this problem before but I wasn't following
and then forgot about it. 😖

cc @nathansobo @maxbrunsfeld

matthewwithanm committed:
Prevent the resize handle from covering the center

Currently, the absolutely positioned element is the only thing with a
size. That means that it'll cover the center. With this change, we give
a size to the (relatively-positioned) atom-dock element so that closed
docks take up canvas space instead.

I think @simurai mentioned this problem before but I wasn't following
and then forgot about it. 😖

simurai reviewed (approved):
Oh.. right. I guess [that comment](https://github.com/atom/atom/pull/13977#discussion_r108034302) got collapsed after a code change.

It's [fixed](https://github.com/atom/one-dark-ui/blob/586f4fdbd386c74f3ab5042f056fb82c67bf9438/styles/docks.less#L13-L19) in the One themes, but much better to inline it so the number stays in sync. 👍

nathansobo commented:
@matthewwithanm These CI failures seem legitimate. Do you think we could release to beta without this today and hotfix it once we get a green build? Seems like we could start getting feedback from beta users.

matthewwithanm commented:
@nathansobo yeah, that seems fine

Commit:
:arrow_up: autocomplete-plus to fix tests

matthewwithanm committed:
Don't show resize handle at all for collapsed docks

They no longer have any functionality. Themes can add extra affordances
if they want.

matthewwithanm committed:
Fix positioning of left dock toggle button when panel container has min-width

matthewwithanm commented:
Button positioning at window sizes when One shows outer padding and when it doesn't:

<img width="1494" alt="screen shot 2017-04-12 at 11 36 45 am" src="https://cloud.githubusercontent.com/assets/126263/24973813/0645648c-1f75-11e7-812c-a0018df32e4d.png">
<img width="847" alt="screen shot 2017-04-12 at 11 36 35 am" src="https://cloud.githubusercontent.com/assets/126263/24973814/07d5551e-1f75-11e7-9d23-294ee7e0a5a8.png">

@simurai I think we'll want to remove the 4px min-width/height from One.

matthewwithanm commented:
oh, One doesn't add the border anymore XD

Here's an updated screenshot taken with atom/one-light-ui#98:

<img width="1311" alt="screen shot 2017-04-12 at 12 12 43 pm" src="https://cloud.githubusercontent.com/assets/126263/24975172/95f850a4-1f79-11e7-8317-eb77690ebc3a.png">

@simurai seriously, great work (:

simurai commented:
> @matthewwithanm what do you think about making the toggle buttons light (`@tool-panel-background-color`?) when the docks are collapsed?

Sure. Actually, now that there is no resize-handle when collapsed, we could absolute position it when open. And then have just a `1px` border. 

Before | After
--- | ---
![screen shot 2017-04-13 at 4 59 54 pm](https://cloud.githubusercontent.com/assets/378023/24995673/a763d68c-206a-11e7-9a62-0ebe099063a8.png) | ![screen shot 2017-04-13 at 4 59 18 pm](https://cloud.githubusercontent.com/assets/378023/24995679/abec94be-206a-11e7-845f-dd3d9eba49af.png)

Looks less heavy and more seamless with the docks. I tried that before, but there were issues when dragging that now seem to be resolved. 🎉 

Already pushed to https://github.com/atom/one-light-ui/pull/98 + https://github.com/atom/one-dark-ui/pull/201. 😇

iKlsR commented:
@matthewwithanm What's that cool sidebar in your editor?

matthewwithanm commented:
@iKlsR that's @suda's tool-bar, populated with a bunch of Nuclide features (:
